backend/src/graphql/resolver/UserResolver.ts

```diff
@@ -224,6 +225,19 @@ export const activationLink = (optInCode: LoginEmailOptIn): string => {
   return CONFIG.EMAIL_LINK_SETPASSWORD.replace(/{optin}/g, optInCode.verificationCode.toString())
 }
 
+const newGradidoID = async (): Promise<string> => {
+  let gradidoId: string
+  let countIds: number
+  do {
+    gradidoId = uuidv4()
+    countIds = await DbUser.count({ where: { gradidoID: gradidoId } })
+    if (countIds > 0) {
+      logger.info('Gradido-ID creation conflict...')
+    }
+  } while (countIds > 0)
+  return gradidoId
+}
+
```

Comment:
I do not understand your point. How can `new User(dbUser)` create a new `gradidoID`? It is only the `dbUser` that may create a new `gradidoID`

Reply:
@Tirokk you are right, that between the initialization of a User with a newGradidoID and the database write access finished per commit, it will be possible that an other user could be created with the same gradidoID, because both users are still not committed till now.
But in the end the first user, which will be written and committed in database wins the game. The commit for the second user will run in an unique key constraint. It is undefined which of the two new users will be commited first.
So i would say, that we should start with the current implementation and collect experiences. There will follow further features with the gradidoID and then we could improve it, if necessary. Ok?